elasticsearch/transport.py

```diff
@@ -339,7 +339,8 @@ def perform_request(self, method, url, headers=None, params=None, body=None):
 
         if headers is None:
             headers = {}
-        headers["user-agent"] = "elasticsearch-py/%s (Python %s)" % (__versionstr__, python_version())
+        if "user-agent" not in headers:
+            headers["user-agent"] = "elasticsearch-py/%s (Python %s)" % (__versionstr__, python_version())
```

Comment:
Could we perhaps not do this at every request? I think this should follow the same exact logic as setting `content-type` at https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/connection/http_urllib3.py#L129 - done only once when instantiating the client and kept there as the default. Also all the headers will then be set in a single place.